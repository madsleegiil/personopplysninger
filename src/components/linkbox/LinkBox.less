.lenkepanel {
  margin-bottom: 0.3rem;
}

.linkbox {
  &__rad {
    display: flex;
    display: flex-end;
  }

  &__tittel {
    margin: 0 0 0.6rem 0;
  }

  &__icon-container {
    padding: 0 1.5rem 0 0;
  }

  &__lenke-container {
    display: flex;
    align-items: center;
  }

  &__content {
    width: 100%;
    display: flex;
    flex-direction: column;
  }
  &__seksjon {
    margin: 0 1rem 0 0;
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    align-items: flex-start;

    @media @tablet {
      width: 50%;
    }

    @media @mobile {
      width: 100%;
    }
  }
}
