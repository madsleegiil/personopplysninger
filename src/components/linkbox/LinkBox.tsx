import React, { useState, useEffect } from "react";
import Mobile from "./responsive/Mobile";
import Desktop from "./responsive/Desktop";

interface Props {
  id: string;
  header: string;
  information: string;
  linkText: string;
  url: string;
  icon?: string;
  infoBoxContent: {
    __html: string;
  };
}

const LinkBox = (props: Props) => {
  const [erMobil, setErMobil] = useState(window.innerWidth <= 420);

  useEffect(() => {
    window.addEventListener("resize", () =>
      setErMobil(window.innerWidth <= 420)
    );
    return () => {
      window.removeEventListener("resize", () =>
        setErMobil(window.innerWidth <= 420)
      );
    };
  });

  return erMobil ? <Mobile {...props} /> : <Desktop {...props} />;
};

export default LinkBox;
