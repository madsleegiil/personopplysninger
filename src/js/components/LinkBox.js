import React from 'react';
import PropTypes from 'prop-types';
import Box from '../components/Box';

const LinkBox = ({
  header, information, url, linkText, kilde,
}) => (
  <Box header={header}>
    <div className="link-box-content">
      <div className="link-info">
        {information}
      </div>
      <a href={url}>{linkText}</a>
      <div className="box-footer link-footer">
        Kilde: {kilde}
      </div>
    </div>
  </Box>
);

LinkBox.propTypes = {
  header: PropTypes.string,
  information: PropTypes.string,
  url: PropTypes.string,
  linkText: PropTypes.string,
  kilde: PropTypes.string,
};

LinkBox.defaultProps = {
  header: '',
  information: '',
  url: '',
  linkText: '',
  kilde: '',
};

export default LinkBox;
