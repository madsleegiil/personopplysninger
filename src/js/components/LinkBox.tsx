/* eslint-disable react/no-danger */
import React, { Component } from "react";
import {
  Undertekst,
  Ingress,
  Undertittel,
  Normaltekst
} from "nav-frontend-typografi";
import Box from "./Box";
import infoIcon from "../../assets/img/infomation-circle.svg";
import InfoBox from "./InfoBox";

interface Props {
  id: string;
  header: string;
  information: string;
  linkText: string;
  url: string;
  kilde?: string;
  icon?: string;
  infoBoxContent: {
    __html: string;
  };
}

interface State {
  displayInfo: boolean;
}

class LinkBox extends Component<Props, State> {
  state = { displayInfo: false };

  render() {
    const {
      id,
      header,
      information,
      url,
      linkText,
      kilde,
      icon,
      infoBoxContent
    } = this.props;

    const { displayInfo } = this.state;

    const toggleInfo = () => {
      this.setState({
        displayInfo: !displayInfo
      });
    };

    return (
      <Box id={id} header={header} icon={icon}>
        {infoBoxContent && (
          <>
            <button
              type="button"
              className="information-circle"
              onClick={() => toggleInfo()}
            >
              <img src={infoIcon} alt="Information" />
            </button>
            {displayInfo && (
              <InfoBox>
                <Undertittel>Informasjon om {header}</Undertittel>
                <Normaltekst className="info-content">
                  <div dangerouslySetInnerHTML={infoBoxContent} />
                </Normaltekst>
              </InfoBox>
            )}
          </>
        )}
        <div className="link-box-content">
          <Ingress>{information}</Ingress>
          <Ingress>
            <a
              className="lenke"
              href={url}
              target="_blank"
              rel="noopener noreferrer"
            >
              {linkText}
            </a>
          </Ingress>
          {kilde && (
            <div className="box-footer link-footer">
              <Undertekst>
                Kilde: {kilde} {/* TODO: må gjøres om til intl */}
              </Undertekst>
            </div>
          )}
        </div>
      </Box>
    );
  }
}

export default LinkBox;
