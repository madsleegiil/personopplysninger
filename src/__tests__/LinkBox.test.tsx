import * as React from "react";
import ReactTestRenderer from "react-test-renderer";
import LinkBox from "../js/components/LinkBox/LinkBox";
import wrapIntl from "../js/IntlTestHelper";

test("render LinkBox with content", () => {
  const link = {
    id: "pensjonsopptjening",
    header: "Pensjon",
    information: "NAV har opplysninger om din pensjon i en egen tjeneste.",
    linkText: "G책 til Din pensjon",
    url: "https://www.nav.no/no/Person/Pensjon/Hva+kan+jeg+fa+i+pensjon",
    kilde: "Pensjonsregisteret",
    infoBoxContent: {
      __html:
        'P책 grunn av tekniske og juridiske begrensninger vil ikke alle dokumenter vises. Du kan ogs책 be om partsinnsyn ved 책 kontakte NAV Kontaktsenter, tlf. 55 55 33 33, eller bruke <a class="lenke" href="" target="_blank" rel="noopener noreferrer">Send beskjed til NAV</a>.'
    }
  };

  const component = ReactTestRenderer.create(
    wrapIntl(
      <LinkBox
        id={link.id}
        header={link.header}
        information={link.information}
        url={link.url}
        linkText={link.linkText}
        kilde={link.kilde}
        infoBoxContent={link.infoBoxContent}
      />
    )
  );
  expect(component.toJSON()).toMatchSnapshot();
});
