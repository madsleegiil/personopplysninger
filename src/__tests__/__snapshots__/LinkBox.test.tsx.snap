// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render LinkBox with content 1`] = `
<div
  className="panel box"
>
  <div
    className="box__header"
    id="pensjonsopptjening"
  >
    <h2
      className="typo-systemtittel"
    >
      Pensjon
    </h2>
  </div>
  <div
    className="box__content"
  >
    <button
      className="information-circle"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Information"
        src={[Function]}
      />
    </button>
    <div
      className="link-box-content"
    >
      <p
        className="typo-ingress"
      >
        NAV har opplysninger om din pensjon i en egen tjeneste.
      </p>
      <p
        className="typo-ingress"
      >
        <a
          className="lenke"
          href="https://www.nav.no/no/Person/Pensjon/Hva+kan+jeg+fa+i+pensjon"
          rel="noopener noreferrer"
          target="_blank"
        >
          GÃ¥ til Din pensjon
        </a>
      </p>
      <div
        className="box-footer link-footer"
      >
        <p
          className="typo-undertekst"
        >
          Kilde: 
          Pensjonsregisteret
           
        </p>
      </div>
    </div>
  </div>
</div>
`;
